oarepo_requests/ui/theme/assets/semantic-ui/js/oarepo_requests_ui/common/utils/hooks.js

```diff
@@ -0,0 +1,246 @@
+import React, { useState, useCallback } from "react";
+import { i18next } from "@translations/oarepo_requests_ui/i18next";
+import { Button, Message, Icon } from "semantic-ui-react";
+import { useMutation } from "@tanstack/react-query";
+import {
+  useCallbackContext,
+  REQUEST_TYPE,
+  WarningMessage,
+  RequestCommentInput,
+} from "@js/oarepo_requests_common";
+import { useFormikContext } from "formik";
+
+/**
+ * @typedef {import("semantic-ui-react").ConfirmProps} ConfirmProps
+ */
+
+const ConfirmationModalConfirmButton = (uiProps) => (
+  <Button
+    className="requests confirmation-modal-confirm-button"
+    content={i18next.t("OK")}
+    {...uiProps}
+  />
+);
+
+const ConfirmationModalCancelButton = (uiProps) => (
+  <Button
+    content={i18next.t("Cancel")}
+    className="requests confirmation-modal-cancel-button"
+    {...uiProps}
+  />
+);
+export const useConfirmDialog = (requestOrRequestType) => {
+  /** @type {[ConfirmProps, (props: ConfirmProps) => void]} */
+  const formik = useFormikContext();
+  const [confirmDialogProps, setConfirmDialogProps] = useState({
+    open: false,
+    content: i18next.t("Are you sure?"),
+    cancelButton: <ConfirmationModalCancelButton />,
+    confirmButton: <ConfirmationModalConfirmButton />,
+    onCancel: () =>
+      setConfirmDialogProps((props) => ({ ...props, open: false })),
+    onConfirm: () =>
+      setConfirmDialogProps((props) => ({ ...props, open: false })),
+  });
+
+  const confirmAction = useCallback(
+    (onConfirm, requestActionType, extraData) => {
+      const dangerous = extraData?.dangerous;
+      /** @type {ConfirmProps} */
+      let newConfirmDialogProps = {
+        open: true,
+        header: `${i18next.t("Are you sure you wish to")} ${
+          requestOrRequestType.name
+        }`,
+        content: dangerous ? <WarningMessage /> : i18next.t("Are you sure?"),
+        onConfirm: () => {
+          setConfirmDialogProps((props) => ({ ...props, open: false }));
+          onConfirm();
+        },
+        onCancel: () => {
+          setConfirmDialogProps((props) => ({ ...props, open: false }));
+          formik?.resetForm();
+        },
+      };
+      switch (requestActionType) {
+        case REQUEST_TYPE.CREATE:
+          newConfirmDialogProps.header = `${i18next.t("Create request")} (${
```

Comment:
I would refactor each case into separate func (return object & spread over conf diag props), to keep confirmAction as short & maintainable as possible